### DISCUSSION TITLE
Status page

### DISCUSSION THREAD

**Discussion by mrpea** [Netflix/dispatch#3446] [Ideas]
Hi - have you discussed internally what it would look like to have a status page as part of Dispatch? I assume you have something at Netflix for internal system status - have you integrated that with Dispatch at all? It would be very useful for us to be able to easily update status from Dispatch incidents.

**Comment by kevgliss** [comment]
Hi @mrpea, I'm not sure if it's documented; but we do have the `default/incidents/status` endpoint that provides an abbreviated view of only active incidents. We typically provide this to leaders so they can quickly get a grasp on the current status of incidents managed by Dispatch. Is this similar to what you're thinking about? Or were you thinking something more akin to the aws health status page: https://health.aws.amazon.com/health/status ?

**Reply by mrpea** [reply]
Thanks for that link - we didn't know about it, that is helpful. 

I'm thinking more like a status page where you can define system components and mark their health (operational, degraded, outage). It would be for the end-users for the system having an incident that you wouldn't normally give the raw incident data to. I think it could be as simple as the list of types of incidents in the project and (insert magic here) the status based on active incidents with severity 1 or 2 and (stretch goal) click to see the incident updates for the open incident.

**Reply by kevgliss** [reply]
Okay, that makes it clearer. We don't really have the concept of "system" that I think we would need. I see what you mean about potentially using "incident type" as a stand-in (e.g. one incident type per system), but I think that make a lot of assumptions and wouldn't be clear to folks. 

I don't think this is something we would be willing to directly contribute to. But I'd happy to talk it through with you in more detail, it it was something you'd be willing to contribute.

**Comment by mrpea** [comment]
Great thank you @kevgliss . Yes it is something I can contribute.

I agree there are a lot of assumptions. I want to keep it as simple as possible to start and then iterate where it makes sense if that works for you.

So - the simplest thing I can think of is a page that lists the incident types and indicates which types have active incidents. I need guidance from you on what to name the endpoint since you have a status endpoint already.

We probably don't want to show incidents that are a very low severity - so maybe we can add an `exclude_from_statuspage` on the incident_severity table? (Or `include_in_statuspage` however you prefer).

In this model each type would display as either Operational or Impacted with a success/checkmark or warning/triangle icon.

Perhaps we can show a simple read-only table (type, severity, title, description) at the top of the page for active incidents.

How does this sound for scope and architecture?
